src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1385,17 +1385,18 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
 
     ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
     uct_dc_mlx5_iface_dci_put(iface, dci_index);
+
+    if (ep->dci == UCT_DC_MLX5_EP_NO_DCI) {
+        uct_dc_mlx5_iface_reset_dci(iface, dci_index);
+    }
+
```

Comment:
if we reset the DCI here, than the error message in uct_dc_mlx5_iface_set_ep_failed which reads WQE content will not be valid
IMO, the fix should be like this:
```
  ...
  uct_dc_mlx5_iface_set_ep_failed()
  if (uct_dc_mlx5_iface_dci_put(..)) {
     assert(ep->dci == NO_DCI)
     uct_dc_mlx5_iface_progress_pending()
  }
} /* end of function */
```

to move uct_dc_mlx5_iface_dci_put() and uct_dc_mlx5_iface_progress_pending() after uct_dc_mlx5_iface_reset_dci